kafka-connect-s3/src/test/java/io/confluent/connect/s3/S3SinkConnectorConfigTest.java

```diff
@@ -227,5 +230,61 @@ public String generatePartitionedPath(String topic, String encodedPartition) {
       }
     }
   }
+
+  @Test
+  public void testConfigurableCredentialProvider() {
+    final String ACCESS_KEY_NAME = "access.key";
+    final String SECRET_KEY_NAME = "secret.key";
+    final String ACCESS_KEY_VALUE = "AKIAAAAAKKKKIIIIAAAA";
+    final String SECRET_KEY_VALUE = "WhoIsJohnGalt?";
+
+    properties.put(
+        S3SinkConnectorConfig.CREDENTIALS_PROVIDER_CLASS_CONFIG,
+        FakeAWSCredentialsProvider.class.getName()
+    );
+    String configPrefix = S3SinkConnectorConfig.CREDENTIALS_PROVIDER_CONFIG_PREFIX;
+    properties.put(configPrefix.concat(ACCESS_KEY_NAME), ACCESS_KEY_VALUE);
+    properties.put(configPrefix.concat(SECRET_KEY_NAME), SECRET_KEY_VALUE);
+    connectorConfig = new S3SinkConnectorConfig(properties);
+
+    AWSCredentialsProvider credentialsProvider = connectorConfig.getCredentialsProvider();
+
+    assertEquals(ACCESS_KEY_VALUE, credentialsProvider.getCredentials().getAWSAccessKeyId());
+    assertEquals(SECRET_KEY_VALUE, credentialsProvider.getCredentials().getAWSSecretKey());
+
+  }
+
+  static class FakeAWSCredentialsProvider implements AWSCredentialsProvider, Configurable {
+    private AWSCredentials credentials;
+
+    @Override
+    public AWSCredentials getCredentials() {
+      return credentials;
+    }
+
+    @Override
+    public void refresh() {
+      throw new UnsupportedOperationException(
+          "Refresh is not supported for this credentials provider");
+    }
+
+    @Override
+    public void configure(final Map<String, ?> configs) {
+
```

Comment:
I know it's just a test class, but given it's the only example we provide here, should we just set two `final` member fields (e.g. `keyId` and `secretKey`) instead of using `configs` from the implemented methods? I feel it's slightly a better pattern.